timemachine/fe/single_topology.py

```diff
@@ -0,0 +1,783 @@
+        all_dummy_c_angle_idxs.extend(all_idxs[5])
+
+        all_dummy_bond_params.extend(all_params[0])
+        all_dummy_angle_params.extend(all_params[1])
+        all_dummy_proper_params.extend(all_params[2])
+        all_dummy_improper_params.extend(all_params[3])
+        all_dummy_x_angle_params.extend(all_params[4])
+        all_dummy_c_angle_params.extend(all_params[5])
+
+    # generate parameters for mol_a
+    mol_a_top = topology.BaseTopology(mol_a, ff)
+    mol_a_bond_params, mol_a_hb = mol_a_top.parameterize_harmonic_bond(ff.hb_handle.params)
+    mol_a_angle_params, mol_a_ha = mol_a_top.parameterize_harmonic_angle(ff.ha_handle.params)
+    mol_a_proper_params, mol_a_pt = mol_a_top.parameterize_proper_torsion(ff.pt_handle.params)
+    mol_a_improper_params, mol_a_it = mol_a_top.parameterize_improper_torsion(ff.it_handle.params)
+
+    mol_a_bond_params = mol_a_bond_params.tolist()
+    mol_a_angle_params = mol_a_angle_params.tolist()
+    mol_a_proper_params = mol_a_proper_params.tolist()
+    mol_a_improper_params = mol_a_improper_params.tolist()
+
+    mol_a_bond_idxs = recursive_map(mol_a_hb.get_idxs(), a_to_c)
+    mol_a_angle_idxs = recursive_map(mol_a_ha.get_idxs(), a_to_c)
+    mol_a_proper_idxs = recursive_map(mol_a_pt.get_idxs(), a_to_c)
+    mol_a_improper_idxs = recursive_map(mol_a_it.get_idxs(), a_to_c)
+
+    all_dummy_bond_idxs = recursive_map(all_dummy_bond_idxs, b_to_c)
+    all_dummy_angle_idxs = recursive_map(all_dummy_angle_idxs, b_to_c)
+    all_dummy_proper_idxs = recursive_map(all_dummy_proper_idxs, b_to_c)
+    all_dummy_improper_idxs = recursive_map(all_dummy_improper_idxs, b_to_c)
+    all_dummy_x_angle_idxs = recursive_map(all_dummy_x_angle_idxs, b_to_c)
+    all_dummy_c_angle_idxs = recursive_map(all_dummy_c_angle_idxs, b_to_c)
+
+    # parameterize the combined molecule
+    mol_c_bond_idxs = mol_a_bond_idxs + all_dummy_bond_idxs
+    mol_c_bond_params = mol_a_bond_params + all_dummy_bond_params
+
+    mol_c_angle_idxs = mol_a_angle_idxs + all_dummy_angle_idxs
+    mol_c_angle_params = mol_a_angle_params + all_dummy_angle_params
+
+    mol_c_proper_idxs = mol_a_proper_idxs + all_dummy_proper_idxs
+    mol_c_proper_params = mol_a_proper_params + all_dummy_proper_params
+
+    mol_c_improper_idxs = mol_a_improper_idxs + all_dummy_improper_idxs
+    mol_c_improper_params = mol_a_improper_params + all_dummy_improper_params
+
+    mol_c_x_angle_idxs = all_dummy_x_angle_idxs
+    mol_c_x_angle_params = all_dummy_x_angle_params
+
+    mol_c_c_angle_idxs = all_dummy_c_angle_idxs
+    mol_c_c_angle_params = all_dummy_c_angle_params
+
+    return (
+        (mol_c_bond_idxs, mol_c_bond_params),
+        (mol_c_angle_idxs, mol_c_angle_params),
+        (mol_c_proper_idxs, mol_c_proper_params),
+        (mol_c_improper_idxs, mol_c_improper_params),
+        (mol_c_x_angle_idxs, mol_c_x_angle_params),
+        (mol_c_c_angle_idxs, mol_c_c_angle_params),
+    )
+
+
+class SingleTopologyV2:
```

Comment:
Would this stay here or be moved to fe.topology?
Would the current SingleTopology be removed?

Reply:
This will be removed - but a lot of files need to be cleaned up (a lot of logic on the MD/estimator needs will need to be modified). I will take care of it during deboggling week.